### PR TITLE
network_custom

### LINKED ISSUE #222: ServiceNetwork subnet can interfere with existing network
The helpernode module currently does not support changing the ServiceNetwork parameter, my environment already uses the 172.30.x.x network, so I cannot use the terraform automation there. Please add network variables for the terraform deployment.

Excerpt of the default install-config yaml ending up on the HelperNode:
networking:
  clusterNetwork:
  - cidr: 10.128.0.0/14 
    hostPrefix: 23 
  networkType: OpenShiftSDN
  serviceNetwork: 
  - 172.30.0.0/16

### PR DESCRIPTION
**Pull request by schabrolles**
Add the possibility to choose the clusterNetwork and serviceNetwork CIDR. (mandatory for multicluster implementation with RHACM and submariner).

This change depends from another PR : https://github.com/ocp-power-automation/ocp4-playbooks/pull/130

/assign @bpradipt

### CODE REVIEW

File: variables.tf
Comment (Prajyot-Parab) [inline variables.tf]: Change default value to `172.30.0.0/16`

**Review by Prajyot-Parab — CHANGES_REQUESTED** [review]

File: docs/var.tfvars-doc.md
Comment (Prajyot-Parab) [inline docs/var.tfvars-doc.md]: We can rename variables to `cluster_network_cidr` and `service_network` (at all places)

File: variables.tf
Comment (Prajyot-Parab) [inline variables.tf]: Change in description needed? (to distinguish from `clusterNetwork_CIDR`)

File: modules/5_install/variables.tf
Comment (Prajyot-Parab) [inline modules/5_install/variables.tf]: No need to specify default values here.

**Review by Prajyot-Parab — CHANGES_REQUESTED** [review]

File: variables.tf
Comment (schabrolles) [reply variables.tf]: good catch ... thanks

File: variables.tf
Comment (schabrolles) [reply variables.tf]: done

**Comment by schabrolles** [comment]
@Prajyot-Parab, changes has been applyed

File: docs/ocp4-manual-deployment-dhcp.md
Comment (Prajyot-Parab) [inline docs/ocp4-manual-deployment-dhcp.md]: This should remain as `serviceNetwork` here as per my understanding.

**Review by Prajyot-Parab — CHANGES_REQUESTED** [review]

File: docs/ocp4-manual-deployment-dhcp.md
Comment (schabrolles) [reply docs/ocp4-manual-deployment-dhcp.md]: yes, totaly correct

**Review by Prajyot-Parab — COMMENTED** [review]
/lgtm (waiting for dependent playbook PR to get merged)

**Comment by schabrolles** [comment]
@Prajyot-Parab  ocp-power-automation/ocp4-playbooks#130 seems to be merged. could you proceed with this PR ?

**Comment by yussufsh** [comment]
@schabrolles please update the hash in the variables.tf file to your commit. Once this PR is merged @Prajyot-Parab  can update the release branches. 
 https://github.com/ocp-power-automation/ocp4-upi-powervm/blob/69c79d7cac77f804f328826c7408673fa9ffe0fe/variables.tf#L278

**Comment by schabrolles** [comment]
Thanks for the clarification @yussufsh

**Comment by schabrolles** [comment]
I just added the hostnetwork prefix customization in "ocp4-playbook". (https://github.com/ocp-power-automation/ocp4-playbooks/pull/136) 

We need to wait for this one to be merged. 

Thanks.

**Comment by schabrolles** [comment]
Hello,

I don't understand what's happening here and why this PR failed verifications

**Comment by mkumatag** [comment]
> Hello,
> 
> I don't understand what's happening here and why this PR failed verifications

I see a merge conflict, can you rebase your changes?

**Comment by schabrolles** [comment]
Rebase against what ? 
can you tell me more about the procedure ? git rebase ???

**Comment by yussufsh** [comment]
> Rebase against what ? can you tell me more about the procedure ? git rebase ???

@schabrolles I hope this will help https://medium.com/@yussufshaikh/power-of-git-rebase-to-resolve-conflicts-from-upstream-7f2aefbd19f

**Comment by schabrolles** [comment]
@Prajyot-Parab, Could you have a look to this PR please ?

File: modules/5_install/install.tf
Comment (Prajyot-Parab) [inline modules/5_install/install.tf:67]: can you please format this section to look uniform, will just need to add additional spaces to rest of vars.

Comment (Prajyot-Parab) [reply modules/5_install/install.tf:67]: just this minor formatting change, rest lgtm.

**Comment by schabrolles** [comment]
@Prajyot-Parab done

**Comment by schabrolles** [comment]
@Prajyot-Parab do you think this PR is now OK to be merged ?

**Comment by schabrolles** [comment]
Somebody can have a look to merge this one ?

**Review by yussufsh — APPROVED** [review]
/lgtm

**Review by Prajyot-Parab — APPROVED** [review]
/lgtm

**Comment by yussufsh** [comment]
@aishwaryabk could you please help cherrypick this commit into the release branches with respective playbook tags?

**Comment by aishwaryabk** [comment]
> @aishwaryabk could you please help cherrypick this commit into the release branches with respective playbook tags?

@yussufsh Sure.

**Pull request merged**
